components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1007,6 +1015,249 @@ public void updateResidentIdP(IdentityProvider identityProvider, String tenantDo
+     *
+     * @param node       filter node.
+     * @param expression list of expression.
+     * @throws IdentityProviderManagementClientException Error when passing invalid filter.
+     */
+    private void setExpressionNodeList(Node node, List<ExpressionNode> expression)
+            throws IdentityProviderManagementClientException {
+
+        if (node instanceof ExpressionNode) {
+            if (((ExpressionNode) node).getAttributeValue().contains(IdPManagementConstants.IDP_IS_ENABLED)) {
+                if ("true".contains(((ExpressionNode) node).getValue())) {
+                    ((ExpressionNode) node).setValue(IdPManagementConstants.IS_TRUE_VALUE);
+                } else if ("false".contains(((ExpressionNode) node).getValue())) {
+                    ((ExpressionNode) node).setValue(IdPManagementConstants.IS_FALSE_VALUE);
+                } else {
+                    String message = "Invalid \'isEnabled\' value passed in the filter. It should be \'true\' or " +
+                            "\'false\' isEnabled = " + ((ExpressionNode) node).getValue();
+                    throw IdPManagementUtil.handleClientException(IdPManagementConstants.ErrorMessage.ERROR_CODE_RETRIEVE_IDP,
+                            message);
+                }
+            }
+            expression.add((ExpressionNode) node);
+        } else if (node instanceof OperationNode) {
+            setExpressionNodeList(node.getLeftNode(), expression);
+            setExpressionNodeList(node.getRightNode(), expression);
+        }
+    }
+
+    /**
+     * Set the passing parameters as result.
+     *
+     * @param limit     page limit.
+     * @param offset    offset value.
+     * @param filter    filter value for IdP search.
+     * @param sortOrder order of IdP(ASC/DESC).
+     * @param sortBy    the column value need to sort.
+     * @param result    result object.
+     * @throws IdentityProviderManagementClientException Error while set offset.
+     */
+    private void setParameters(int limit, int offset, String filter, String sortOrder, String sortBy, IdpSearchResult
+            result) throws IdentityProviderManagementClientException {
+
+        result.setLimit(validateLimit(limit));
+        result.setOffSet(validateOffset(offset));
+        result.setSortBy(validateSortBy(sortBy));
+        result.setSortOrder(validateSortOrder(sortOrder));
+        result.setFilter(filter);
+    }
+
+    /**
+     * Validate sortBy.
+     *
+     * @param sortBy sortBy attribute.
+     * @return Validated sortOrder and sortBy.
+     */
+    private String validateSortBy(String sortBy) {
+
+        if (StringUtils.isBlank(sortBy)) {
+            if (log.isDebugEnabled()) {
+                log.debug("sortBy attribute is empty. Therefore we set the default sortBy attribute. sortBy" +
+                        IdPManagementConstants.DEFAULT_SORT_BY);
+            }
+            return IdPManagementConstants.DEFAULT_SORT_BY;
+        } else {
```

Comment:
else part can get out of the else statement. 

        if (StringUtils.isBlank(sortBy)) {
            if (log.isDebugEnabled()) {
                log.debug("sortBy attribute is empty. Therefore we set the default sortBy attribute. sortBy" +
                        IdPManagementConstants.DEFAULT_SORT_BY);
            }
            return IdPManagementConstants.DEFAULT_SORT_BY;
        } 

            switch (sortBy) {
                case IdPManagementConstants.IDP_NAME:
                    sortBy = IdPManagementConstants.NAME;
                    break;
                case IdPManagementConstants.IDP_HOME_REALM_ID:
                    sortBy = IdPManagementConstants.HOME_REALM_ID;
                    break;
                default:
                    sortBy = IdPManagementConstants.DEFAULT_SORT_BY;
                    if (log.isDebugEnabled()) {
                        log.debug("sortBy attribute is incorrect. Therefore we set the default sortBy attribute. " +
                                "sortBy: " + IdPManagementConstants.DEFAULT_SORT_BY);
                    }
                    break;
            }
            return sortBy;

Reply:
This is done